src/app/highchart/highchart.component.ts

```diff
@@ -251,4 +278,24 @@ export class HighchartComponent implements OnInit {
       this.chart.tooltip.refresh([level.points[latestLevel], ytd.points[latestYtd]]);
     }
   }
+
+  findLastValue(valueArray) {
+    let counter = valueArray.length;
+    while (counter-- && !valueArray[counter].y);
```

Comment:
Can `y` every be 0? If the last value is 0, the counter will still be decremented.